src/App.tsx

```diff
@@ -1,55 +1,77 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { debounce } from 'lodash';
+import cn from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { Autocomplete } from './components/Autocompete';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPersonSlug, setSelectedPersonSlug] = useState('');
+  const [query, setQuery] = useState('');
+  const [debouncedQuery, setDebouncedQuery] = useState('');
+  const [isTyping, setIsTyping] = useState(false);
+  const selectedPerson: Person | undefined = peopleFromServer
+    .find(person => person.slug === selectedPersonSlug);
+
+  const applyQuery = useCallback(
+    debounce(setDebouncedQuery, 1000),
+    [],
+  );
+
+  const onSelect = (slug: string) => {
+    setSelectedPersonSlug(slug);
+  };
+
+  const applyTypingDelay = useCallback(
+    debounce(setIsTyping, 1000),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setIsTyping(true);
+    applyQuery(event.target.value);
+    applyTypingDelay(false);
+  };
+
+  useEffect(() => {
+    setQuery('');
+    setDebouncedQuery('');
+    setIsTyping(false);
+  }, [selectedPerson]);
+
+  const dropDownList = peopleFromServer.filter(person => (
```

Comment:
I couldn't find the implementation of the following requirement:

don't run filtering again if the text has not changed (pause in typing happened when the text was the same as before)

this can be achieved by using useEffect while filtering so it filters only when debounced query changes.